packages/lesswrong/components/comments/CommentsNewForm.tsx

```diff
@@ -284,8 +298,51 @@ const CommentsNewForm = ({
     if (formDisabledDueToRateLimit) {
       submitBtnProps.disabled = true
     }
-    
+
+    const { DialogueCheckBox } = Components
+
+    const targetUserId = parentComment?.userId
+    const targetUserDisplayName = parentComment?.user?.displayName
+
+    const {
+      userDialogueChecksResult: { results: userDialogueChecks = [] },
+    } = useDialogueMatchmaking({ getMatchedUsers: false, getRecommendedUsers: false, getOptedInUsers: false, getUserDialogueChecks: true });
+
+    const { checkId, userIsChecked, userIsMatched } = parentComment?.user ? getUserCheckInfo(parentComment?.user, userDialogueChecks) : { checkId: null, userIsChecked: null, userIsMatched: null }
+    console.log(document, targetUserId, targetUserDisplayName, checkId, userIsChecked, userIsMatched)
+
+    const { LWHelpIcon, LWTooltip } = Components
+
+    const test = useABTest(showMatchBoxesInCommentReplyFormABTest)
+
+    const topLevelCommentId = parentComment?.topLevelCommentId ?? parentComment?._id
+
+    const { count } = useMulti({
+      terms: {view: "oneReplyInThreadByUser", topLevelCommentId, userId: currentUser?._id},
+      collectionName: "Comments",
+      fragmentName: 'CommentId',
+      skip: !topLevelCommentId || !currentUser,
+    });
+
+    const showDialogueCheckbox = targetUserId && targetUserDisplayName && test === "experiment" && count && count > 0
```

Comment:
Is there forum gating applied to this at any point? We don't want this to appear on the forum atm